Detekt to 1.7.4

cortinico opened the pull request:
##  :page_facing_up: Context
Let's update Detekt before #298 

## :pencil: Changes
Seems like our detekt configuration was still using an older version (`1.4.0`). I've extracted that versions to use the version defined at the top level (same for `ktlint).

## :stopwatch: Next steps
I've re-gerated the config file with `./gradlew detektGenerateConfig`. I've also copied over as we can potentially turn on more rules that are currently disabled by default.

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+
+comments:
+  active: true
+  excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Chucker doesn't use Spek.

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    threshold: 4
+  StringLiteralDuplication:
+    active: false
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Same thing about Spek.

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    ignoreStringsRegex: '$^'
   TooManyFunctions:
     active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Same thing about Spek.

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    methodNames: 'toString,hashCode,equals,finalize'
+  InstanceOfCheckForException:
+    active: false
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Same thing about Spek.

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    active: false
+  TooGenericExceptionCaught:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Same thing about Spek.

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+formatting:
+  active: true
+  android: false
+  autoCorrect: true
```
We are turning on autoCorrect feature?

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    active: false
+    autoCorrect: true
+    indentSize: 4
+    continuationIndentSize: 4
```
It seems that these rules for formatting already have some code style. Should we export it and use for #249 ?

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+  active: true
+  ClassNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    classPattern: '[A-Z$][a-zA-Z0-9$]*'
+  ConstructorParameterNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    ignoreOverridden: true
+  EnumNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    enumEntryPattern: '^[A-Z][_a-zA-Z0-9]*'
+  ForbiddenClassName:
+    active: false
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    forbiddenName: ''
+  FunctionMaxLength:
+    active: false
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    maximumFunctionNameLength: 30
+  FunctionMinLength:
+    active: false
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    minimumFunctionNameLength: 3
+  FunctionNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    ignoreOverridden: true
+  FunctionParameterNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    ignoreOverridden: true
+  ObjectPropertyNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    privatePropertyPattern: '(_)?[A-Za-z][_A-Za-z0-9]*'
+  PackageNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    packagePattern: '^[a-z]+(\.[a-z][A-Za-z0-9]*)*$'
+  TopLevelPropertyNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    privatePropertyPattern: '_?[A-Za-z][_A-Za-z0-9]*'
+  VariableMaxLength:
+    active: false
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    maximumVariableNameLength: 64
+  VariableMinLength:
+    active: false
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    minimumVariableNameLength: 1
+  VariableNaming:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    active: true
+  ForEachOnRange:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
+  SpreadOperator:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+potential-bugs:
+  active: true
+  Deprecation:
+    active: false
```
Let's turn it on.

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    active: true
+  LateinitUsage:
+    active: false
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    maxJumpCount: 1
+  MagicNumber:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    active: true
+    ignoreOverridableFunction: true
+    excludedFunctions: 'describeContents'
+    excludeAnnotatedFunction: 'dagger.Provides'
```
We don't have Dagger in Chucker)

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    acceptableDecimalLength: 5
+  UnnecessaryAbstractClass:
+    active: true
+    excludeAnnotatedClasses: 'dagger.Module'
```
We don't have Dagger in Chucker :)

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+    active: false
+  WildcardImport:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
```
Redundant Spek mentions

vbuberen commented on detekt-config.yml:
```diff
@@ -1,27 +1,615 @@
+  WildcardImport:
+    active: true
+    excludes: '**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt'
+    excludeImports: 'java.util.*,kotlinx.android.synthetic.*'
```
We can not exclude these. At least `synthetic`, since it will be a good catch for people who forget to use `ViewBinding`, like happened in one of recent PRs.

cortinico commented:
Answering here for all the comments. What I added is the default Detekt config file. I had to add it as I removed the `buildUponDefaultConfig` flag. 

If we start customising it extensively (e.g. removing all the `Spec` exclusions, etc.), we will have to manually re-apply those changes whenever we regenerate the default config file (with newer rules, etc.).

If you wish, I can also revert to use the `buildUponDefaultConfig` flag and just have our customisation. It's just a bit harder to discover new rules, but not a big deal.

vbuberen commented:
I thought you want to be able to do a in-depth customisation and decided to add such default file.
If we are not going to change this default config file when I would just revert to our previous approach, since there is no sense in keeping it.

cortinico commented:
> I thought you want to be able to do a in-depth customisation and decided to add such default file.

Yeah that was the idea. But I wanted to do this in a incremental way, like in one or more subsequent PRs as I don't know how many Rules failure I have to fix once I start enabling them.

cortinico commented:
Anyway if you wish we can also take the `buildUponDefaultConfig` approach and keep only our customisation.

vbuberen commented:
Let's do these incremental changes after releasing 3.2.0 and just merge the update now with old way of configuration.
